app/models/user.rb

```diff
@@ -1,16 +1,27 @@
 class User < ApplicationRecord
+  after_create :set_roles
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable, :confirmable,
+  # :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :confirmable, :registerable,
          :recoverable, :rememberable, :validatable
-  has_many :likes
-  has_many :posts
-  has_many :comments
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
 
   validates :name, presence: true
-  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0, allow_nil: true }
 
-  def recents_posts
-    posts.limit(3).order(created_at: :desc)
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def admin?
+    role == 'admin'
+  end
```

Comment:
- [ ] Great work adding this method in your users model 👏🏿 . However, remember it should return true or false. For example: `role == 'admin' ? true: false`

Reply:
Based on this role == 'admin' will return true or false without even the function because ruby can check for itself what it returns and based on the solutions I've found online role == 'admin' is also correct